src/Core/Startup/ErrorControlChainMiddleware.php

```diff
@@ -88,8 +94,19 @@ protected function safeReloadWithToken(HTTPRequest $request, $reloadToken)
         $request->getSession()->init($request);
 
         // Next, check if we're in dev mode, or the database doesn't have any security data, or we are admin
-        if (Director::isDev() || !Security::database_is_ready() || Permission::check('ADMIN')) {
-            return $reloadToken->reloadWithToken();
+        $reload = function (HTTPRequest $request) use ($reloadToken) {
+            $env = $this->getApplication()->getKernel()->getEnvironment();
+            if ($env === Kernel::DEV || !Security::database_is_ready() || Permission::check('ADMIN')) {
+                return $reloadToken->reloadWithToken();
+            }
+            return null;
+        };
+
+        // Process middleware
+        $chainWithMiddleware = static::create($this->getApplication());
```

Comment:
This is the part of this logic I'm least happy with... I have to construct a second DI version of ErrrorControlChainMiddlware here.

I'd have used Director::singleton() instead, but callMiddleware isn't public.